### PR TITLE
Fixes #4185: Added Dark Mode for HelpActivity , FaqListActivity , FaqSingleActivity

### LINKED ISSUE #4185: Add dark mode support to HelpActivity, FaqListActivity and FaqSingleActivity
Add dark mode support to HelpActivity, FaqListActivity and FaqSingleActivity including the contained views.

Go through this [wiki](https://github.com/oppia/oppia-android/wiki/Dark-Mode) to get started.

### PR DESCRIPTION
**Pull request by bhaktideshmukh**
## Explanation
Fixes #4185 
Added dark mode for HelpActivity , FaqListActivity , FaqSingleActivity.

Mocks Link

- FaqList :   https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/c14812a2-e4b5-4c02-8001-7ea1bb146aa3/specs/
- FaqSingle : https://xd.adobe.com/view/c05e9343-60f6-4c11-84ac-c756b75b940f-950d/screen/d13800b0-9b6e-431b-93da-c4374e60947e/specs/

**ScreenShots**

| Default mode  | After dark mode|
|-------|---|
|||

| Default mode | After dark mode |
|-------|---|
|||

| Default mode  | After dark mode |
|-------|---|
|||

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

### CODE REVIEW

**Comment by ayush0402** [comment]
@bhaktideshmukh Haven't looked at the changes yet but also update the color for ActionBar to the correct color. You might want to look at the theme in the manifest that this screen is using and refer to OppiaActionBarTheme we are currently using for another theme for this.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh Haven't looked at the changes yet but also update the color for ActionBar to the correct color. You might want to look at the theme in the manifest that this screen is using and refer to OppiaActionBarTheme we are currently using for another theme for this.

Do you mean to update 'android:theme="@style/Widget.AppCompat.ActionBar" ' to 'android:theme="@style/OppiaThemeWithoutActionBar" in  the toolbar section.

**Comment by rt4914** [comment]
Un-assigning until approved / re-assigned by @ayush0402

**Comment by ayush0402** [comment]
@bhaktideshmukh We are not supposed to change anything directly in the manifest file. If you look at the current declaration of `OppiaTheme` it will have something like `<item name="actionBarTheme">@style/OppiaActionBarTheme</item>` to set the ActionBar color. 
But I guess FAQ Activity is not actually using ActionBar but instead using a custom Toolbar in its layout file. So we just might need to change that.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh We are not supposed to change anything directly in the manifest file. If you look at the current declaration of `OppiaTheme` it will have something like `<item name="actionBarTheme">@style/OppiaActionBarTheme</item>` to set the ActionBar color. But I guess FAQ Activity is not actually using ActionBar but instead using a custom Toolbar in its layout file. So we just might need to change that.

I have updated the theme in Faq_list_activity.xml -- is this what you suggested?

current view:--
||

**Comment by bhaktideshmukh** [comment]
PTAL @ayush0402 
Thanks

File: app/src/main/res/layout-sw600dp-land/faq_single_activity.xml
Comment (ayush0402) [inline app/src/main/res/layout-sw600dp-land/faq_single_activity.xml]: This should be a component_color.

File: app/src/main/res/layout/faq_list_activity.xml
Comment (ayush0402) [inline app/src/main/res/layout/faq_list_activity.xml]: ditto.

File: app/src/main/res/layout/faq_single_activity.xml
Comment (ayush0402) [inline app/src/main/res/layout/faq_single_activity.xml]: ditto.

**Review by ayush0402 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Thanks, overall this looks good. Just had a few suggestions.
Also, I think the content of the FAQ and other list items in FAQ questions list should be using `sub_text color` (which is defined in color_defs as `color_def_oppia_silver`.) and not the Body Header color, you might need to confirm this from the mocks and let me know if It is actually mentioned otherwise in the mocks.

File: app/src/main/res/layout-sw600dp-land/faq_single_activity.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-sw600dp-land/faq_single_activity.xml]: Done

File: app/src/main/res/layout/faq_single_activity.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/faq_single_activity.xml]: > @bhaktideshmukh Thanks, overall this looks good. Just had a few suggestions. Also, I think the content of the FAQ and other list items in FAQ questions list should be using `sub_text color` (which is defined in color_defs as `color_def_oppia_silver`.) and not the Body Header color, you might need to confirm this from the mocks and let me know if It is actually mentioned otherwise in the mocks.

Content of FAQ in mocks is mentioned as 'oppia_primary_text'.

**Comment by BenHenning** [comment]
@ayush0402 I'm deferring to you before taking a pass (i.e. I'll wait until you approve this before I'll review). Ditto for the other dark mode PRs.

@bhaktideshmukh I think @ayush0402 made some comments above, is this ready for another review pass?

**Comment by bhaktideshmukh** [comment]
> @ayush0402 I'm deferring to you before taking a pass (i.e. I'll wait until you approve this before I'll review). Ditto for the other dark mode PRs.
> 
> @bhaktideshmukh I think @ayush0402 some comments above, is this ready for another review pass?

Yes I made the changes suggested by @ayush0402  on this and my other PR on dark mode back then only.

**Comment by bhaktideshmukh** [comment]
@rt4914 Please review this PR as well.

**Comment by rt4914** [comment]
Blocking review because the divider should be visible in light and dark mode for Help activity. Keeping it assigned to myself as what steps we can take depends upon the review of other PRs.

File: app/src/main/res/values/component_colors.xml
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: component_color_faq_activity_toolbar_color

File: app/src/main/res/values/component_colors.xml
Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: component_color_faq_activity_container_background_color

Or you can directly use `component_color_faq_activity_background_color ` in whichever xml file its required

Comment (rt4914) [inline app/src/main/res/values/component_colors.xml]: component_color_faq_activity_container_background_color

Or you can directly use `component_color_faq_activity_background_color ` in whichever xml file its required

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh In screenshots dividers are not visible. Mostly because you have removed the drawable. Mostly @Akshatkamboj14 's PRs should help how he fixed the divider issue and once his PRs gets merged I think you won't need to fix it specifically.

**Comment by bhaktideshmukh** [comment]
> @bhaktideshmukh In screenshots dividers are not visible. Mostly because you have removed the drawable. Mostly @Akshatkamboj14 's PRs should help how he fixed the divider issue and once his PRs gets merged I think you won't need to fix it specifically.

This divider issue is still pending.

File: app/src/main/res/values/component_colors.xml
Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

File: app/src/main/res/values/component_colors.xml
Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

File: app/src/main/res/layout/help_item.xml
Comment (rt4914) [inline app/src/main/res/layout/help_item.xml:16]: Use `general_item_border` instead of `general_item_background_border `

and `general_item_border_cyan` instead of `general_item_background_border_cyan`

Comment (bhaktideshmukh) [reply app/src/main/res/layout/help_item.xml:16]: Done, divider issue is fixed now.
Updated the screenshots in the PR description as well.

File: app/src/main/res/layout/faq_single_activity.xml
Comment (Akshatkamboj14) [inline app/src/main/res/layout/faq_single_activity.xml]: @bhaktideshmukh, in the FAQ activity the text color is not correct, according to the mocks it should be **#C4C4C4** not def white, PTAL.

File: app/src/main/res/layout/faq_content.xml
Comment (Akshatkamboj14) [inline app/src/main/res/layout/faq_content.xml:41]: I think you should change the **oppia_faq_divider** to **color_def_dark_silver** as we are shifting colors from **colors_migrating.xml** to **color_defs.xml**, Thanks.

**Review by Akshatkamboj14 — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/faq_single_activity.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/faq_single_activity.xml]: @Akshatkamboj14 Do we need to consider the mock colors or the colors which are presently used in develop branch for light mode?
Currently in develop branch def_white color is used for textColor.
@rt4914 please confirm whether the mock colors should be used or colors of develop branch.

File: app/src/main/res/layout/faq_content.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/faq_content.xml:41]: > I think you should change the **oppia_faq_divider** to **color_def_dark_silver** as we are shifting colors from **colors_migrating.xml** to **color_defs.xml**, Thanks.

Develop uses oppia_faq_divider, although I tried changing the color and this is how it turned out to be--
| oppia_faq_divider  | color_def_dark_silver  |
|-------|---|
|||

File: app/src/main/res/layout/faq_single_activity.xml
Comment (rt4914) [reply app/src/main/res/layout/faq_single_activity.xml]: By Looking at different mocks I can confirm that `#C4C4C4` is incorrect in mocks. That colour is used for secondary text but here these texts are primary and therefore current color is right.

File: app/src/main/res/layout-sw600dp-land/faq_single_activity.xml
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/faq_single_activity.xml]: In all these colors I think they should be renamed by replacing 

`....faq_acitivty` with `...faq_single_activity`. If you need to create new colors thats fine.

File: app/src/main/res/layout/faq_content.xml
Comment (rt4914) [inline app/src/main/res/layout/faq_content.xml]: If this file is getting used in `FAQSingleActivity` then rename this colors too as per above comment.

**Review by rt4914 — CHANGES_REQUESTED** [review]
@bhaktideshmukh Please separate FAQListActivity colors from `FAQSingleActivity` colors correctly.

File: app/src/main/res/layout/faq_content.xml
Comment (Akshatkamboj14) [reply app/src/main/res/layout/faq_content.xml:41]: No Bhakti **oppia_faq_divider** and **color_def_dark_silver** have the same hex codes so I think there should be no changes in UI.

File: app/src/main/res/values/component_colors.xml
Comment (Akshatkamboj14) [inline app/src/main/res/values/component_colors.xml]: Optional Changes - you can add space between the comments as it should be better because all other files' comments have spaces between them.

Comment (bhaktideshmukh) [reply app/src/main/res/values/component_colors.xml]: Done

File: app/src/main/res/layout/faq_content.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/faq_content.xml]: Done

File: app/src/main/res/layout-sw600dp-land/faq_single_activity.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout-sw600dp-land/faq_single_activity.xml]: Done

File: app/src/main/res/layout/faq_content.xml
Comment (bhaktideshmukh) [reply app/src/main/res/layout/faq_content.xml:41]: Here it was my fault in naming the color I apologize for that and now I have fixed that.

Comment (rt4914) [reply app/src/main/res/layout/faq_content.xml:41]: No worries. Thanks.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

File: app/src/main/res/values/component_colors.xml
Comment (rt4914) [reply app/src/main/res/values/component_colors.xml]: Confirmed.

**Pull request merged**
